### PR TITLE
Telemetry features + docs

### LINKED ISSUE #216: Easier way to enable fallback reporting?
It would be nice to be able to do something like `duckdb::fallback_reporting(TRUE)`. I think that could write a file to `tools::R_user_dir("duckplyr", "cache")`/`DUCKPLYR_FALLBACK_LOG_DIR` that has lower precedence than the environment variables.

I think requiring the user to edit their `~/.Renviron` is too big of a barrier to data collection, and it would be nice to reduce:

```
The duckplyr package is configured to fall back to dplyr when it encounters an incompatibility. Fallback events can be collected and
uploaded for analysis to guide future development. By default, no data will be collected or uploaded.
ℹ Fallback logging is not controlled and therefore disabled. Enable it with [Sys.setenv(DUCKPLYR_FALLBACK_COLLECT = 1)](vscode-file://vscode-app/Applications/Positron.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#), disable it with
  [Sys.setenv(DUCKPLYR_FALLBACK_COLLECT = 0)](vscode-file://vscode-app/Applications/Positron.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#).
ℹ Fallback uploading is not controlled and therefore disabled. Enable it with [Sys.setenv(DUCKPLYR_FALLBACK_AUTOUPLOAD = 1)](vscode-file://vscode-app/Applications/Positron.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#), disable it
  with [Sys.setenv(DUCKPLYR_FALLBACK_AUTOUPLOAD = 0)](vscode-file://vscode-app/Applications/Positron.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#).
ℹ See [duckplyr::fallback](vscode-file://vscode-app/Applications/Positron.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#) for details.
```

to something like:

```
duckplyr falls back to dplyr when it encounters an incompatibility. 
We would like to find out about these events so we can make duckplyr better.
i Use `duckdb::fallback_upload()` to upload this report
i Use `duckdb::fallback_reporting(TRUE)` to automatically upload future reports
```

I'm not sure why we're not logging by default, since that seems like a very low risk activity?

### LINKED ISSUE #338: Notes on telemetry
- Once the vignette exists should it be linked from the README in a badge that'd say "help us" or so.
- Instead of the Sys.setenv stuff should there be an exported helper function to change the settings?

### PR DESCRIPTION
**Pull request by maelle**
Fix #216 
Fix #338

### CODE REVIEW

**Comment by maelle** [comment]
verbose by default is horrible in the tests

File: R/fallback.R [hunk 1]
```diff
@@ -13,34 +13,23 @@
...
 #'
-#' - \code{DUCKPLYR_FALLBACK_AUTOUPLOAD} controls uploading, set it
-#'   to 1 or greater to enable uploading.
-#'   If the value is 0, uploading is disabled.
-#'   Currently, uploading is active if the value is 1 or greater.
-#'   Future versions of duckplyr may start logging additional data
-#'   and thus require a higher value to enable uploading.
-#'   Set to 99 to enable uploading for all future versions.
-#'   Use [usethis::edit_r_environ()] to edit the environment file.
+#' - To opt the current machine in automatic report uploads, run [`fallback_reporting_optin()`].
```
Comment (krlmlr) [inline R/fallback.R:30]: It's not the machine, it's the user config:

```suggestion
#' - To opt in to automatic report uploads, run [`fallback_reporting_optin()`].
```

File: R/fallback.R [hunk 2]
```diff
@@ -105,16 +93,20 @@ fallback_sitrep <- function() {
...
   if (is.na(fallback_uploading)) {
-    c("i" = "Fallback uploading is not controlled and therefore disabled. Enable it with {.run Sys.setenv(DUCKPLYR_FALLBACK_AUTOUPLOAD = 1)}, disable it with {.run Sys.setenv(DUCKPLYR_FALLBACK_AUTOUPLOAD = 0)}.")
+    c(
+      i = "Use {.fun duckplyr::fallback_upload} to upload this report and existing reports.",
+      i = "Use {.fun duckplyr::fallback_reporting_optin} to automatically upload future reports.",
+      i = "Use {.fun duckplyr::fallback_review} to review existing reports."
+    )
   } else if (fallback_uploading) {
-    c("v" = "Fallback uploading is enabled.")
+    c("v" = "Fallback automatic uploading is on. Disable it with {.fun duckplyr::fallback_reporting_optout}.")
```
Comment (krlmlr) [inline R/fallback.R:107]: Automatic fallback?

File: R/fallback.R [hunk 3]
```diff
@@ -141,19 +133,6 @@ fallback_txt_help <- function() {
   )
 }
 
-fallback_nudge <- function(call_data) {
```
Comment (krlmlr) [inline R/fallback.R:144]: Why are we losing this?

**Review by krlmlr — COMMENTED** [review]
What issue is this PR related to?

Why are we now changing to configuration based on the existence of files?

**Comment by maelle** [comment]
@krlmlr I updated the first comment with two issue references.

File: R/fallback.R (hunk 3 above)
Comment (maelle) [reply R/fallback.R:144]: to simplify. We advertise in other places and if the docs are clearer (they are not clearer yet) then no nudge is needed.

**Comment by krlmlr** [comment]
In the end, I reworked info and logging and moved a lot of things around in #426 and prior PRs. The configuration is stored in a single DCF file, loaded and applied during package load by setting environment variables that aren't yet set externally. The `.Renviron` and other means of setting environment vars take precedence.

**Pull request closed without merging**